TLM/TLM/Manager/Impl/AdvancedParkingManager.cs

```diff
@@ -63,7 +63,8 @@ public class AdvancedParkingManager
                     case ExtPathMode.DrivingToTarget: {
                         // citizen instance requires a car but is walking: release instance to
                         // prevent it from floating
-                        if ((Singleton<CitizenManager>.instance.m_instances.m_buffer[(ushort)citizenInstanceId].m_flags & CitizenInstance.Flags.Character) != 0) {
+                        ref CitizenInstance citizenInstance = ref ((ushort)citizenInstanceId).ToCitizenInstance();
```

Comment:
Would it be worth adding an extension that accepts `this uint citizenInstanceId` and automatically casts to `ushort` internally?

```csharp
internal static ref CitizenInstance ToCitizenInstance(this uint citizenInstance)
    => ref _citizenInstanceBuffer[(ushort)citizenInstance];
```

Reply:
Yeah, probably.

Reply:
mmmh shit ... now that I thought about it and double checked, this casting actually suxx bad and could lead to errors.
valueoutside_ushort casted to ushort will always produce a ushort, even for values outside ushort. it will just overflow and by default overflow is not checked :/
So we can either do the checked {} thing, which will cost us some perf, or we can just hope for the best, in which case it does not matter if I do cast or don't at all it seems to me :/

Reply:
Ok so I added the uint extension method but without the cast.
As far as I can tell array indexes are always int. In .net framework for sure, in Mono I did not find the code XD
Which means that all values given to it will be coverted to int anyway.
Precasting it would cause errors in case we give it a value > ushort.MaxValue and we get an overflow.
So I'd rather rely on an IndexOutOfRangeException in case we mess up instead getting an overflow, not realizing it and manipulating the wrong data.
Please correct me if you think my logic is wrong :D